### PR TITLE
fix for #1808

### LINKED ISSUE #1808: [BUG] Mailbox contents not deleted
# Bug Report

## Context
Mailbox contents remain in `maildata` subdirectory after running `sudo ./setup.sh email del test@example.com`.

### What is affected by this bug?
Deleting mailboxes.
### When does this occur?
When deleting a mailbox with the `setup.sh` script.
### How do we replicate the issue?

```
sudo ./setup.sh email del test@example.com
Password: 
sed: can't read /tmp/docker-mailserver/postfix-virtual.cf: No such file or directory
Do you want to delete the mailbox as well(all mails will be removed)?(y/n) y

Mailbox deleted.
```
Check contents of `maildata` subdirectory.

### Actual Behavior
The mailbox is no longer listed when running `sudo ./setup.sh email list`. But the `test` mailbox and its contents are still in the `maildata` subdirectory. 
### Expected Behavior
The mailbox contents should have been deleted.
## Your Environment
Synology DS918+ with latest DSM and Docker version 18.09.8, build bfed4f5.

### Environment Variables

```yml
---
version: "2"

# Define services
services:
  
  roundcubemail:
    image: roundcube/roundcubemail:latest
    container_name: roundcubemail
    restart: unless-stopped
    volumes:
      - /v/d/roundcube/www:/var/www/html
      - /v/d/roundcube/sqlite:/var/roundcube/db
    ports:
      - 127.0.0.1:8000:80
    environment:
      - ROUNDCUBEMAIL_DB_TYPE=sqlite
      - ROUNDCUBEMAIL_SKIN=elastic
      - ROUNDCUBEMAIL_DEFAULT_HOST=ssl://mail.example.com
      - ROUNDCUBEMAIL_SMTP_SERVER=ssl://mail.example.com
      - ROUNDCUBEMAIL_DEFAULT_PORT=993
      - ROUNDCUBEMAIL_SMTP_PORT=465
  
  solr:
    container_name: mailserver_solr
    image: lmmdock/dovecot-solr:latest
    volumes:
      - /v/d/solr-dovecot:/opt/solr/server/solr/dovecot
    restart: always
  
  mailserver:
    image: docker.io/mailserver/docker-mailserver:latest
    hostname: mail
    domainname: example.com
    container_name: mailserver
    ports:
      - "465:465"
      - "993:993"
    volumes:
      - /v/d/mailserver/maildata:/var/mail
      - /v/d/mailserver/mailstate:/var/mail-state
      - /v/d/mailserver/maillogs:/var/log/mail
      - /v/d/mailserver/config:/tmp/docker-mailserver/
      - /v/d/ssl/output:/tmp/ssl:ro
      - /v/d/mailserver/solr-dovecot/10-plugin.conf:/etc/dovecot/conf.d/10-plugin.conf:ro
    environment:
      - SSL_TYPE=manual
      - SSL_CERT_PATH=/tmp/ssl/fullchain.pem
      - SSL_KEY_PATH=/tmp/ssl/privkey.pem
      - POSTFIX_MESSAGE_SIZE_LIMIT=102400000
      - ENABLE_SPAMASSASSIN=0
      - ENABLE_CLAMAV=0
      - ENABLE_FAIL2BAN=0
      - ENABLE_POSTGREY=0
      - ENABLE_SASLAUTHD=0
      - ONE_DIR=1
      - DMS_DEBUG=0
    cap_add:
      - NET_ADMIN
      - SYS_PTRACE
    restart: always
```
## Workaround?
Contents of mailbox are deleted when running `sudo docker exec mailserver rm -R /var/mail/example.com/test/` afterwards.

### PR DESCRIPTION
**Pull request by aendeavor**
# Description

Bugfix and small improvement for `setup.sh`

Fixes #1808

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation (README.md or ENVIRONMENT.md or the Wiki)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by aendeavor** [comment]
Please not that the deletion of mail users now strictly follows the `setup.sh help` text, deleting only one user at a time. This is intended and does not actually introduce changes technically, since the documentation always stated that it was exactly this way.

File: target/bin/addmailuser [hunk 1]
```diff
@@ -47,8 +47,11 @@ touch "${DATABASE}"
 (
   flock -e 200
 
-  grep -qi "^$(escape "${USER}")|" "${DATABASE}" 2>/dev/null &&
-    errex "User \"${USER}\" already exists"
+  if grep -qi "^$(escape "${USER}")|" "${DATABASE}" 2>/dev/null
+  then
+    echo "User '${USER}' already exists."
+    exit 0
```
Comment (casperklein) [inline target/bin/addmailuser:53]: `exit 1` should fit better IMO.

File: target/bin/delmailuser [hunk 2]
```diff
@@ -90,73 +90,72 @@ fi
...
-  do
-    # very simple plausibility check
-    [[ ${USER} != *'@'*'.'* ]] && errex "No valid address: ${USER}"
+  # very simple plausibility check
+  [[ ${USER} != *@*.* ]] && errex "No valid address: ${USER}"
 
-    declare -a MAILARR
-    MAILARR[0]="${USER%@*}"
-    MAILARR[1]="${USER#*@}"
+  unset MAILARR
```
Comment (casperklein) [inline target/bin/delmailuser:98]: Why is that needed? MAILARR is not set before?

**Comment by casperklein** [comment]
Instead of array elements, why not using normal vars that are more readable/intuitive to understand? 😎

Something like this:

```
USER="${*}"               -->  EMAIL="${*}"

MAILARR[0]="${USER%@*}"   -->  USER="${USER%@*}"
MAILARR[1]="${USER#*@}"   -->  DOMAIN=${USER#*@}"
```

**Comment by wernerfred** [comment]
But then using `EMAIL` and not `USER` ;)

File: target/bin/delmailuser (hunk 2 above)
Comment (aendeavor) [reply target/bin/delmailuser]: In my tests in #1808 the array was (for some reason) already filled.

If I remember correctly, the unset of some variables in the current version (in this PR) should not interfere with something else since it's locked behind the subshell access (?). I'm deliberately unsetting `EMAIL USER DOMAIN` even though it might strictly speaking not be necessary.

File: target/bin/addmailuser (hunk 1 above)
Comment (aendeavor) [reply target/bin/addmailuser:53]: This will make `setup.sh` print an unexpected error. I found this to be rather misleading, but I can revert back to `exit 1` if you want.

File: target/bin/delmailuser (hunk 2 above)
Comment (casperklein) [reply target/bin/delmailuser]: if `MAILARR` is already filled, but not expected to be, we should find out why.

Only this two files are also involved.
```
setup.sh
helper-functions.sh
```
In none of them I could find a reference for `MAILARR`. Can you double check, that `MAILARR` is really set?

File: target/bin/addmailuser (hunk 1 above)
Comment (casperklein) [reply target/bin/addmailuser:53]: If an email address cannot be added, because it's already existing, it's an (unexpected) error. Therefore `exit 0` feels odd to me. But I'll leave it up to you, it's not worth to discuss this one to death 😆

Comment (casperklein) [reply target/bin/addmailuser:53]: Ok. I just saw, you got rid of the MAILARR altogether 👍

Comment (aendeavor) [reply target/bin/addmailuser:53]: > Ok. I just saw, you got rid of the MAILARR altogether +1

Belong to the other comment I guess :D

I will revert back to `exit 1` and resolve this conversation:)

File: target/bin/delmailuser (hunk 2 above)
Comment (aendeavor) [reply target/bin/delmailuser]: As you mentioned, I got rid of it altogether. I think we can resolve this?

Comment (casperklein) [reply target/bin/delmailuser]: Why this:  `unset EMAIL USER DOMAIN` ?

File: target/bin/addmailuser (hunk 1 above)
Comment (casperklein) [reply target/bin/addmailuser]: Yes, I mixed things up. Sorry.

File: target/bin/delmailuser (hunk 2 above)
Comment (aendeavor) [reply target/bin/delmailuser]: Probably not actually needed. Shall I remove it?

Comment (casperklein) [reply target/bin/delmailuser]: Yes

Comment (casperklein) [reply target/bin/delmailuser]: I guess, add/delete an email address is covered by bats tests? Otherwise, a quick test setup to see if add/delete works as intended wouldn't hurt.

**Review by casperklein — APPROVED** [review]
